rclrs/src/wait.rs

```diff
@@ -46,7 +46,7 @@ impl Drop for rcl_wait_set_t {
     fn drop(&mut self) {
         // SAFETY: No preconditions for this function (besides passing in a valid wait set).
         let rc = unsafe { rcl_wait_set_fini(self) };
-        if let Err(e) = to_rcl_result(rc) {
+        if let Err(e) = to_rclrs_result(rc) {
```

Comment:
Yes, but isn't using `to_rclrs_result` nicer here, since this function fetches the error message from rcutils?

Using `try_into()/try_from()` also makes this call site more complicated. It would have to be something like this:
```rust
        match RclReturnCode::try_from(rc) {
            Err(e) => panic!("Unable to release WaitSet. Unknown error code {:?}", e),
            Ok(e) if e != RclReturnCode::Ok => panic!("Unable to release WaitSet. {:?}", e),
        }
```

Reply:
This looks much nicer and more idiomatic as it leverages `TryInto`:

```rust

if let Err(e) = TryInto::<RclReturnCode>::try_into(rc) {
    panic!("Unable to release WaitSet. {:?}", e)
}
```

And then we can get rid of `to_rclrs_result` entirely and incorporate it into `TryFrom`

Reply:
This does not do what you mean. `TryInto` is just `Err` if  the error code is unknown.

Reply:
I see what you mean. It somehow feels like a code smell that we have this function outside all the other conversion traits and only used in one specific place. If we can find a way to refactor this in the future, I'd be much happier. For now that's ok.

Reply:
Yes, I also hope we can solve this more elegantly. Thanks for reviewing!